src/engine/SCons/Spirent/__init__.py

```diff
@@ -12,14 +12,10 @@
 
 def vulcan_builder(fs, options, graph):
     # Get the vulcan command line options
-    vulcan_options = options.vulcan_opts
-    if not vulcan_options:
-        options_str = os.environ.get("SCONS_VULCAN_OPTS")
-        if not options_str:
-            print("scons: error, no options given to vulcan command")
-            return 1
-        else:
-            vulcan_options.append(options_str)
+    vulcan_options = options.vulcan_opts or os.environ.get("SCONS_VULCAN_OPTS")
+    if vulcan_options is None:
+        print("scons: error, no options given to vulcan command")
```

Comment:
I had to retain the error message and return here. Other wise it is throwing TypeError for while processing the Subprocess command when no options are given for vulcan command

Reply:
The type error is because `vulcan_options = None` in that case and its is being used as a string. You could fix this with:

`vulcan_opts = options.vulcan_ots or os.environ.get(...) or ""`